Model/ReachNotification.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+namespace Reach\Payment\Model;
+
+use Reach\Payment\Api\Data\ResponseInterface;
+use Reach\Payment\Api\NotificationInterface;
+use Magento\Framework\Webapi\Exception;
+use Magento\Framework\Phrase;
+use Magento\Sales\Api\OrderRepositoryInterface;
+
+/**
+ * ReachNotification model
+ *
+ */
+class ReachNotification implements NotificationInterface
+{
+    /** @var ResponseInterface */
+    protected $response;
+
+    /**
+     * @var \Reach\Payment\Helper\Data
+     */
+    protected $reachHelper;
+
+    /**
+     * @var \Magento\Sales\Model\OrderFactory
+     */
+    protected $_orderFactory;
+
+    /**
+     * @var OrderRepositoryInterface
+     */
+    protected $orderRepository;
+
+    public function __construct(
+        ResponseInterface                 $response,
+        \Reach\Payment\Helper\Data        $reachHelper,
+        \Magento\Sales\Model\OrderFactory $orderFactory,
+        OrderRepositoryInterface          $orderRepository
+    )
+    {
+        $this->response = $response;
+        $this->reachHelper = $reachHelper;
+        $this->_orderFactory = $orderFactory;
+        $this->orderRepository = $orderRepository;
+    }
+
+    /**
+     * @param $request
+     * @param $signature
+     * @return ResponseInterface
+     */
+    public function handleNotification($request, $signature)
+    {
+        try {
+            if ($this->validate($request, $signature)) {
+                // TODO:: Handle other Notification order states
+                $response = json_decode($request, true);
+                if (isset($response['OrderState']) && $response['OrderState'] == "Processed" && $response['UnderReview'] === false && !count($response['Refunds'])) {
```

Comment:
could we make this line more readable? it's a little long and also we could wait to do this until we get a linter/checkstyle in place